src/main/java/mate/academy/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,77 @@
+package mate.academy.dao.impl;
+
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.JoinType;
+import jakarta.persistence.criteria.Root;
+import java.util.Optional;
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.User;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart add(ShoppingCart shoppingCart) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(shoppingCart);
```

Comment:
```suggestion
            session.merge(shoppingCart);
```

or saveOrUpdate() - read common mistakes file

Reply:
At first, I used the persist() method because save is deprecated since 6.0, but I changed it to the save() method after reading the checklist and understanding the problem. But why is merge() or saveOrUpdate() better in this cas, because save() can create copy of User entity?

Reply:
you already said "save" is deprecated, cause it doesn't follow JPA spec

Reply:
I also have a question, why then persist() is used in the add() method in both MoviesessionDao and TicketDao?